### ISSUE TITLE
BG2 (modded): spellbook can't handle 7-character spell icon names

### ISSUE THREAD

**Issue by Tomsod** [gemrb/gemrb#1743]
So, aTweaks has a new Gate icon as part of "PnP Fiends" component, I believe.  It's installed as `fl#905c.bam` for the in-spellbook version.  But this code mishandles the filename as it expects it to be 8 characters long:
https://github.com/gemrb/gemrb/blob/0fe43af76537e80dfc137315e4efa8fbde84a7e1/gemrb/plugins/SPLImporter/SPLImporter.cpp#L124-L128
And the engine ends up with `fl#905cc.bam` which doesn't actually exist.  Presumably it worked for the original game's engine, otherwise the mod's authors would have noticed, so just adding/replacing the 8th character is evidently wrong.  What was this code supposed to do, again?  Replacing the last character (7th or 8th or whatever) works for me, but this particular spell name already ends with 'c', so maybe it should just check for that, instead.

**Comment by lynxlynxlynx** [comment]
Yep, the last char should be a c, nothing more to it. You can use `("{:.{}}c", s->SpellbookIcon, desiredLength)` to parametrize where to cut off. But it needs to be clamped, so we don't create unterminated buffers.

**Comment by Tomsod** [comment]
This is what I did:
```cpp
s->SpellbookIcon[s->SpellbookIcon.length()-1] = 'c';
```
Why is `Format()` necessary again?  Seems overcomplicated, and I think `ReadResRef()` already adds `\0` at the end.

**Comment by lynxlynxlynx** [comment]
Yep, that's simpler, please submit a PR.

**Comment by bradallred** [comment]
Yeah, you don't need to worry about an unterminated buffer since `SpellbookIcon` is a `ResRef`. You're also correct that `ReadResRef` (well any `ResRef` really) will have a null terminator. You're also correct that you dont need `Format` if the only thing you need is to replace the last character with `c`; `*SpellbookIcon .rbegin() = 'c'` ought to do it.

**Comment by Tomsod** [comment]
Yes, that works.  Reverse iterators are neat!
